aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -347,6 +369,28 @@ module aptos_framework::multisig_account {
             num_rejections >= multisig_account_resource.num_signatures_required
     }
 
+    #[view]
+    /// Return true if the owner can execute the "rejected" transaction with given transaction id now.
+    public fun can_reject(
+        owner: address, multisig_account: address, sequence_number: u64): bool acquires MultisigAccount {
+        let multisig_account_resource = borrow_global<MultisigAccount>(multisig_account);
+        assert_is_owner_addr(owner, multisig_account_resource);
+        assert!(
+            sequence_number > 0 && sequence_number < multisig_account_resource.next_sequence_number,
+            error::invalid_argument(EINVALID_SEQUENCE_NUMBER),
+        );
+        let transaction = table::borrow(&multisig_account_resource.transactions, sequence_number);
```

Comment:
This logic seems a bit redundant across can_execute, can_reject and the actual execute function. Can you see if these can be combined

Reply:
Refactored and simplified.